### PR TITLE
[202211][ppi]: Implement port bulk comparison logic

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

**DEPENDS:**
1. https://github.com/sonic-net/sonic-sairedis/pull/1171
2. https://github.com/sonic-net/sonic-sairedis/pull/1197
3. https://github.com/sonic-net/sonic-swss/pull/2806

**HLD:** https://github.com/sonic-net/SONiC/pull/1084

**What I did**
* Implemented port bulk add/remove comparison logic
* Refactored Port OA to facilitate port config operations

**Why I did it**
* To improve switch Fast Boot startup time

**How I verified it**
* UT tests
* VS tests

**Details if related**
* Backport from `master`: https://github.com/sonic-net/sonic-swss/pull/2564

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@nazariig  pls followup on checker failures so once the master PR is merged we can also merge this one

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -3238,548 +3334,622 @@ void PortsOrch::doPortTask(Consumer &consumer)
              * 2. Create new ports
              * 3. Initialize all ports
              */
-            if (m_portConfigState == PORT_CONFIG_RECEIVED || m_portConfigState == PORT_CONFIG_DONE)
+            if (getPortConfigState() != PORT_CONFIG_MISSING)
             {
+                std::vector<PortConfig> portsToAddList;
+                std::vector<sai_object_id_t> portsToRemoveList;
+
+                // Port remove comparison logic
```
Comment (nazariig) [inline orchagent/portsorch.cpp:3342]: extend the decsription

File: tests/mock_tests/portsorch_ut.cpp [hunk 2]
```diff
@@ -12,6 +12,9 @@
 #include "pfcactionhandler.h"
 #include <sys/mman.h>
 #undef private
+#define private public
+#include "warm_restart.h"
+#undef private
```
Comment (saiarcot895) [inline tests/mock_tests/portsorch_ut.cpp:17]: Rather than doing this hack, can friendship declarations be used instead?

**Review by saiarcot895 — APPROVED** [review]
Approved with a comment.

**Review by prgeor — APPROVED** [review]
LGTM.
@nazariig @liat-grozovik please attach the test results for the manual test done to verify AN/LT,FEC, port breakout

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**